src/base/bittorrent/session.cpp

```diff
@@ -3246,17 +3291,23 @@ void Session::setMaxUploads(int max)
     }
 }
 
-bool Session::isUTPEnabled() const
+BTProtocol Session::btProtocol() const
 {
-    return m_isUTPEnabled;
+    return m_btProtocol;
 }
 
-void Session::setUTPEnabled(bool enabled)
+void Session::setBTProtocol(BTProtocol protocol)
 {
-    if (enabled != m_isUTPEnabled) {
-        m_isUTPEnabled = enabled;
-        configureDeferred();
-    }
+    const int value = static_cast<int>(protocol);
+    const int lower = static_cast<int>(BTProtocol::Both);
+    const int upper = static_cast<int>(BTProtocol::UTP);
```

Comment:
I think `lower` and `upper` should be `static` too.

Reply:
took suggestion in https://github.com/qbittorrent/qBittorrent/pull/7446#discussion_r139278144